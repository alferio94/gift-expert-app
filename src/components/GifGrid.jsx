import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) =>
{
    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && <p>Loading...</p>
            }
            {
                !isLoading &&
                <div className="card-grid ">
                    {images.map(image => <GifGridItem key={image.id} {...image} />)}
                </div>
            }
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
