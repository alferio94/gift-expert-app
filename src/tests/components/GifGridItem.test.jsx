import { render, screen, configure } from '@testing-library/react';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem.jsx>', () =>
{
    const title = 'Titulo del GIF';
    const url = 'https://media1.giphy.com/media/3o7btLwJW1QqQqQqZS/200w.gif';
    test('should match with the snapshot', () =>
    {
        const { container } = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('should show the image and the alt', () =>
    {
        render(<GifGridItem title={title} url={url} />);
        /*expect(screen.getByRole('img').src).toBe(url);*/
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('should show the title', () =>
    {
        render(<GifGridItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
});