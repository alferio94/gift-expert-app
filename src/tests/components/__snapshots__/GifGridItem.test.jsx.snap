// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem.jsx> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo del GIF"
      src="https://media1.giphy.com/media/3o7btLwJW1QqQqQqZS/200w.gif"
    />
    <p>
      Titulo del GIF
    </p>
  </div>
</div>
`;
